### ISSUE TITLE
GRAD CAM for test images

### ISSUE THREAD

**Issue by SaraPiresNobrega** [AlexeyAB/darknet#7845]
Hello ! 
I am developing my master's degree dissertation and I would like to used GRAD CAM or other class activation maps to corroborate the test results of my object detection model (YOLOv4). 

Using the information present in https://github.com/AlexeyAB/darknet/issues/5117 it is possible to do that ?

**Comment by AlexeyAB** [comment]
@SaraPiresNobrega Hi,

Yes, use this in cfg-file for Grad-CAM:
```
[net]
adversarial_lr=0.5
attention=1
burn_in=0
learning_rate=0.001
```

And remove all `stopbackward=` lines from cfg-file (if they exist).

And run training command with your pre-trained weights.

----

For example:

* Use this cfg-file (rename it from `yolov4x-mish.cfg.txt` to `yolov4x-mish.cfg`): [yolov4x-mish.cfg.txt](https://github.com/AlexeyAB/darknet/files/6736380/yolov4x-mish.cfg.txt)
* Download this weights: https://github.com/AlexeyAB/darknet/releases/download/darknet_yolo_v4_pre/yolov4x-mish.weights
* You can un-comment these lines and recompile to save outputs to png-files: 
    * https://github.com/AlexeyAB/darknet/blob/97ccee56283ad92118153f0cd1036d1ea07ba115/src/network_kernels.cu#L242-L246
    * https://github.com/AlexeyAB/darknet/blob/97ccee56283ad92118153f0cd1036d1ea07ba115/src/network_kernels.cu#L253-L254
* Run training: `./darknet detector train F:/MSCOCO/coco_f.data cfg/yolov4x-mish.cfg yolov4x-mish.weights`



You will see such results: 

Although it looks less beautiful and less understandable, it is closer to the truth than other Grad-CAM implementations.
![image](https://user-images.githubusercontent.com/4096485/123865609-cc0f1780-d934-11eb-9de9-1c900ec6ec99.png)

**Comment by JeremyKeusters** [comment]
Thanks for this @AlexeyAB ! Very interesting to look at! Would it be easy to also have the labels displayed on these images?

**Comment by SaraPiresNobrega** [comment]
@AlexeyAB thank you very much ! This approach will be very helpful in my master's degree dissertation !

**Comment by AlexeyAB** [comment]
Also try the latest commit, I fixed a bug recently.

**Comment by SaraPiresNobrega** [comment]
@AlexeyAB thank you very much !
